app/controllers/experiments_controller.rb

```diff
@@ -630,4 +633,22 @@ def sort_my_modules(records, sort)
       records
     end
   end
+
+  def generate_two_by_two_connections(src_tasks, dest_tasks)
+    return if src_tasks.empty? || dest_tasks.empty?
+
+    src_tasks.each do |src|
```

Comment:
The MyModule model already has these relation methods defined:
```
  has_many :my_modules, through: :outputs, source: :to, class_name: 'MyModule'
  has_many :my_module_antecessors, through: :inputs, source: :from, class_name: 'MyModule'
```

so we can just use these here

Reply:
Hi Martin,

Thank you for mentioning that.
Will work on it right away.